packages/viz/src/lib/Layer.ts

```diff
@@ -85,28 +86,20 @@ export class Layer {
     const styleField =
       this._style && this._style.field ? [this._style.field] : undefined;
 
-    await this._source.init(styleField);
-
-    const metadata = this._source.getMetadata();
-
-    const styleProps = this._style
-      ? this._style.getProperties(this._source)
-      : undefined;
-
-    // Get properties of the layer
-    const props = this._source.getProps();
+    if (!this._source.isInitialized) {
+      await this._source.init(styleField);
+    }
 
-    const layerProperties = Object.assign(
-      this.id,
-      props,
-      defaultStyles(metadata.geometryType),
-      styleProps
-    );
+    const layerProperties = await this._getLayerProperties();
 
     // Create the Deck.gl instance
     if (this._source instanceof CARTOSource) {
-      this._deckLayer = new MVTLayer(layerProperties);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+      // @ts-ignore
+      this._deckLayer = new MapboxLayer(layerProperties);
```

Comment:
If we use this it will break compatibility with Google Maps since Google Maps doesn't offer way to draw a custom Layer. We need to use the native class of Deck.gl